app/src/test/java/ch/hikemate/app/model/route/HikeTest.kt

```diff
@@ -165,4 +165,103 @@ class HikeTest {
     assertEquals(elevationGain, detailed.elevationGain, deltaForDoubleComparison)
     assertEquals(difficulty, detailed.difficulty)
   }
+
+  @Test
+  fun segmentsWorksAsExpected() {
+    // Case 1: Empty waypoints list
+    val emptyHike =
+        createHike(
+                description = "description",
+                bounds = Bounds(0.0, 0.0, 0.0, 0.0),
+                waypoints = emptyList(),
+                elevation = emptyList(),
+                distance = 0.0,
+                estimatedTime = 0.0,
+                elevationGain = 0.0,
+                difficulty = HikeDifficulty.EASY)
```

Comment:
Nitpick: this is repeated quite a few times, maybe create a `val` at the beginning of the function and use `myVal.copy(waypoints = ...)` to change the waypoints only (because they are the only thing that matters here).

Reply:
Done